src/ansys/dynamicreporting/core/adr.py

```diff
@@ -0,0 +1,250 @@
+import os
+from pathlib import Path
+import re
+import sys
+from typing import Any, Optional, Type
+
+from django.core import management
+from django.http import HttpRequest
+
+from .adr_utils import get_logger
+from .exceptions import (
+    AnsysVersionAbsentError,
+    DatabaseMigrationError,
+    ImproperlyConfiguredError,
+    InvalidAnsysPath,
+    InvalidPath,
+    StaticFilesCollectionError,
+)
+
+
+class ADR:
+    """
+    >> from ansys.dynamicreporting.core import ADR, Table
+    >> opts = { "CEI_NEXUS_DEBUG" : "0", "CEI_NEXUS_SECRET_KEY":"", "CEI_NEXUS_LOCAL_DB_DIR":r"C:\\cygwin64\\home\vrajendr\\ogdocex" }
+    >> adr = ADR(r"C:\\Program Files (x86)\\ANSYSv231", opts = opts)
+    >> adr.configure()
+    >> table = adr.create_item(Table, name="table1", content={}, tags="dp=1 part=bumper")
+    """
+
+    def __init__(
+        self,
+        ansys_installation: str,
+        *,
+        db_directory: str = None,
+        media_directory: str = None,
+        static_directory: str = None,
+        opts: dict = None,
+        request: HttpRequest = None,
+        logfile: str = None,
+    ) -> None:
+        self._db_directory = None
+        self._media_directory = None
+        self._static_directory = None
+
+        if ansys_installation is None:
```

Comment:
@viseshrp In PyEnSight, there is a staticmethod in locallauncher.py that we rather like for this.  get_cei_install_directory().   Tends to allow most folks to ignore the install path.

Reply:
done, thanks